pkg/cmd/openshift-sdn-node/kube_proxy.go

```diff
@@ -119,11 +118,7 @@ func newProxyServer(config *kubeproxyconfig.KubeProxyConfiguration, client clien
 			return nil, fmt.Errorf("unable to read IPTables MasqueradeBit from config")
 		}
 
-		var localDetector proxyutiliptables.LocalTrafficDetector
-		localDetector, err = getLocalDetector(config, iptInterface)
-		if err != nil {
-			return nil, fmt.Errorf("unable to create proxier: %v", err)
-		}
+		localDetector := getLocalDetector()
```

Comment:
ok so we are going to by default just use the interface detector for SDN and that way we don't have to even check if CIDRs are multiple or single.

q: This would be something that's done on startup right? how would it look during upgrades? The iptables rules we add for the previous version of cidr detector gets removed and re-added as these new rules? I don't think we have code to remove anything, so I am assuming upon restart of sdn pods on upgrades, the rules are all recreated freshly?

Reply:
or maybe for existing services it would still be the old rules and newer services it would be the new rules?

Reply:
the kube-proxy code overwrites the previous service rules every time it updates